api/python/provisioner/commands/sw_update.py

```diff
@@ -0,0 +1,244 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+import logging
+from typing import Type
+
+from . import (CommandParserFillerMixin, _ensure_update_repos_configuration,
+               _pre_yum_rollback, _update_component, _apply_provisioner_config
+               )
+from .check import Check
+
+# TODO consider to use RunArgsUpdate and support dry-run
+from .. import inputs
+from ..config import LOCAL_MINION
+from ..errors import (ClusterMaintenanceEnableError, SWStackUpdateError,
+                      ClusterMaintenanceDisableError, HAPostUpdateError,
+                      ClusterNotHealthyError, SWUpdateError, SWUpdateFatalError
+                      )
+from ..hare import (ensure_cluster_is_healthy, consul_export,
+                    cluster_maintenance_disable, apply_ha_post_update,
+                    cluster_maintenance_enable
+                    )
+from ..salt import YumRollbackManager, cmd_run as salt_cmd_run
+
+from ..salt_master import config_salt_master, ensure_salt_master_is_running
+from ..salt_minion import config_salt_minions
+from ..vendor import attr
+
+logger = logging.getLogger(__name__)
+
+
+def _consul_export(stage):
+    # TODO make that configurable to turn off if not needed
+    logger.info(f'Exporting consul [{stage}]')
+    consul_export(fn_suffix=stage)
+
+
+def _restart_salt_minions():
+    logger.info("Restarting salt minions")
+    salt_cmd_run(
+        'systemctl restart salt-minion', background=True
+    )
+
+
+# TODO consider to use RunArgsUpdate and support dry-run
+@attr.s(auto_attribs=True)
+class SWUpdate(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+
+    def run(self, targets):  # noqa: C901 FIXME
+        # logic based on https://jts.seagate.com/browse/EOS-6611?focusedCommentId=1833451&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1833451  # noqa: E501
+
+        checker = Check()
+        check_res = checker.run(targets=LOCAL_MINION)
```

Comment:
That block should be moved inside the following try block. Actually `ensure_cluster_is_healthy` check at the beginning it is exactly the same but with less deepness.

Also I believe it's not safe to make all checks mandatory since the new logic is very new and not enough tested BUT it may breaks SW update logic in false positives cases.

So, I would suggest:
- do all checks
- but do not fail and just post on INFO level if everything is passed and on WARNING level if anything is failed

Reply:
Done. Isolate this logic in separate `SWUpdateDecisionMaker` class.